modules/heat_conduction/src/actions/MortarGapHeatTransferAction.C

```diff
@@ -0,0 +1,349 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MortarGapHeatTransferAction.h"
+
+#include "AddVariableAction.h"
+#include "FEProblem.h"
+#include "libmesh/string_to_enum.h"
+#include "NonlinearSystem.h"
+
+#include "ModularGapConductanceConstraint.h"
+#include "GapFluxModelRadiation.h"
+#include "GapFluxModelConduction.h"
+
+// Counter for modular user objects
+static unsigned int thermal_action_userobject_radiation_counter = 0;
+static unsigned int thermal_action_userobject_conduction_counter = 0;
+
+registerMooseAction("HeatConductionApp", MortarGapHeatTransferAction, "append_mesh_generator");
+registerMooseAction("HeatConductionApp", MortarGapHeatTransferAction, "add_mortar_variable");
+registerMooseAction("HeatConductionApp", MortarGapHeatTransferAction, "add_constraint");
+registerMooseAction("HeatConductionApp", MortarGapHeatTransferAction, "add_user_object");
+
+InputParameters
+MortarGapHeatTransferAction::validParams()
+{
+  InputParameters params = Action::validParams();
+  params.addClassDescription(
+      "Action that controls the creation of all of the necessary objects for "
+      "calculation of heat transfer through an open/closed gap using a mortar formulation and a "
+      "modular design approach");
+
+  params.addParam<Real>("thermal_lm_scaling",
+                        1.,
+                        "Scaling factor to apply to the thermal Lagrange multiplier variable");
+
+  params += ModularGapConductanceConstraint::validParams();
+  params += GapFluxModelRadiation::validParams();
+  params += GapFluxModelConduction::validParams();
+
+  params.makeParamNotRequired<SubdomainName>("primary_subdomain");
+  params.makeParamNotRequired<SubdomainName>("secondary_subdomain");
+  params.makeParamNotRequired<Real>("gap_conductivity");
+  params.addParam<bool>(
+      "user_provided_user_objects",
+      true,
+      "Whether user creates user objects in the input file and provides their block names");
+
+  params.addParam<MooseEnum>("gap_flux_options",
```

Comment:
If you use a `MultiMooseEnum` you only need the `CONDUCTION` and `RADIATION` options. The user can supply none, either, or both `gap_flux_options = 'CONDUCTION RADIATION'`